### ISSUE TITLE
Illegal IDL union accepted by tao_idl

### ISSUE THREAD

**Issue by jwillemsen** [DOCGroup/ACE_TAO#2279]
The following IDL is illegal but accepted by tao_idl, ridlc rejects it with `IDL::ParseError: 'default' case label superfluous for IDL::AST::Union UnionTypeTest`

```
module Test {
enum TypeKind {
STRING_TYPE,
LONG_TYPE,
FLOAT_TYPE
};

union UnionTypeTest switch(TypeKind) {
case STRING_TYPE:
    string string_data;
case LONG_TYPE:
    long long_data;
case FLOAT_TYPE:
    float float_data;
default:
    boolean default_data;
};

};
```

**Comment by jwillemsen** [comment]
Coming from https://github.com/OpenDDS/OpenDDS/discussions/4782

**Comment by iguessthislldo** [comment]
It's a weird case, but I'm not seeing something in the IDL spec that explicitly disallows this.

**Comment by iguessthislldo** [comment]
>  A union type can contain a default label only where the values given in the non-default labels do not cover the entire range of the union's discriminant type.

This? I see it in 4.2 now, it moved into the bottom of the first "Note" in the union section. Alright, so it does explicitly disallow this, assuming "range" is all the valid enum values and not just the underlying integer type.

**Comment by mitza-oci** [comment]
Right, so I guess I just didn't find it in 4.2.  The union discriminator is defined by the enum type, not the underlying type.
